### ISSUE TITLE
Migrating to nextcloud?

### ISSUE THREAD

**Issue by Emporea** [pulsejet/memories#183]
Are there plans to migrate this repo to nextcloud / get the official label and maybe even replace the nextcloud photos at some point, because honestly memories is so much better since it uses metadata and is so much faster..

**Comment by pulsejet** [comment]
> Are there plans to migrate this repo to nextcloud / get the official label

No. At least not as long as I'm the major contributor / maintainer, and not employed by Nextcloud (very unlikely). Also see #12 

I'd leave this open though.

**Comment by meichthys** [comment]
@pulsejet I don't think there are any specified requirements currently. According to [this](https://github.com/nextcloud/twofactor_webauthn/issues/179#issuecomment-1141072423), it's just an admin setting on the app store itself.

**Comment by luxzg** [comment]
I'd love some spotlight for Memories... Least I could do is add my comments, and open another issue in the First run wizard repo:

https://github.com/nextcloud/firstrunwizard/issues/911

**Comment by pulsejet** [comment]
Pinning this because it gets asked so often. I want to clarify the current position on this (from my side):

**Will Memories replace Photos or merge into one app?**

**Short Answer:**
No. Memories has a different target audience, and is meant to provide an alternative to Photos for users who want a better experience. It is not a fork of Photos, and will not be merged upstream.

**Long Answer:**

There are two distinct arguments why the separate apps makes sense.

1. **The target audience may be different.** Photos is included with every installation of Nextcloud and is quite simple and works out-of-the-box, without adding any extra maintenance and database overhead. This situation may be ideal in some cases, e.g. if you're a university and storing photos isn't your primary use case. I can imagine such instances make a large portion of Nextcloud users since it is, after all, an enterprise productivity suite.
2. **Difference in architectural opinions.** The way Memories and Photos work are completely different, and while I believe Memories has "better" design choices, this is not universal opinion. For instance (note that all of these are trade-offs; my presentation may come off as a bit biased):
    - Use of WebDAV: Photos exclusively uses WebDAV for all its operations while Memories uses a JSON REST API. While WebDAV brings some nice things like being able to use any compatible app to view your albums etc. I don't believe it is possible to optimize it enough (especially the way it is implemented) to have acceptable performance. Also WebDAV is an aging standard and uses XML, which is both slow to parse compared to JSON and much harder to make type-safe without runtime checks. And it's got lexical bits.
    - Querying methodology: Photos uses WebDAV for queries, which means philosophically all queries must be formed from the same basic building blocks. The up side is that this makes the implementation more modular and possibly easier to debug. But this makes it impossible (IMO) to do the complex DB queries that are for instance used by the Memories timeline. Instead, Photos performs most functions either in PHP or the frontend, which means the database can't optimize anything.
    - Metadata: Since Photos exclusively (and deliberately) relies on metadata extracted by the Nextcloud core, it means depending on the best that PHP extensions can do (since this is the lightest way). Again this is understandable from a maintenance POV, but it means limited file support, no motion photos, no editing of EXIF and needing to wait for everything to be implemented in the core (which is understandably slow).
    - There are others. But the idea is, each app makes design decisions that don't agree and there's no clear "right" way to make such an app.

**Is there some middle ground?**

One potential solution that I hope for is, Nextcloud can provide better visibility to Memories as an _alternative_ to Photos. A large fraction of Nextcloud administrators are unaware of the existence of Memories or are (understandably) apprehensive of installing a third-party app. IMHO this just hurts Nextcloud's image for home users, evident by all the hate at `r/selfhosted` especially whenever Nextcloud gets mentioned for photo management.

As of now, I've seen a few posts from others mentioning something similar with no response from Nextcloud GmbH. I do understand this is somewhat of an uncomfortable topic though, since Memories may seem like a "competitor" to Photos which was developed by Nextcloud. As a result, I won't push this issue myself, but feel free to upvote any of the existing requests out there if you agree with me :)
